### DISCUSSION TITLE
Congregation WebDAV Sync Removal/Replacement?

### DISCUSSION THREAD

**Discussion by advenimus** [sircharlo/meeting-media-manager#2637] [Q&A] [answered]
Hello Friends,

Just curious to find out if I'm understanding the deprecation notice correctly. 

With the version being released next month, will there no longer be any functionality to sync congregation media from one device to another, such as the current way? (ie. is WebDav being replaced with something else?)

Please let me know, thanks!

**Comment by Supamav** [comment]
Would love to know this, we use this heavily for our meetings.  Both congregations in our hall use it, without it, it presents us with a challenge to be able to prep the meetings in a timely manner.

Thanks so much, for the hard work that goes into this amazing program!

**Reply by e-Feitosa** [reply]
It is a resource that I like very much, I know many congregations that I know use this resource and I consider essential. But in the other posts, the developer, responds to comments about the new summer that is an advanced feature and that many have difficulty using or do not know how to use (some like this). I particularly appreciate this feature and will make it difficult. Even using alternatives like Dropbox, Google Drive to sync home media with KH’s PC, many will have difficulties to include the media in the new M3. **But let’s see if the developer (@sircharlo ) reconsiders including this great feature. I really would like it to stay available.**

_Read more about this subject at the link:_ 
https://github.com/sircharlo/mmm-refactor/discussions/2

**Comment by sircharlo** [comment, marked as answer]
> Hello Friends,
> 
> Just curious to find out if I'm understanding the deprecation notice correctly.
> 
> With the version being released next month, will there no longer be any functionality to sync congregation media from one device to another, such as the current way? (ie. is WebDav being replaced with something else?)
> 
> Please let me know, thanks!

Good day @advenimus and @Supamav!
In short, your understanding is correct. 

Quite frankly, it has become very time-consuming to both find free and reputable WebDAV providers and to play whack-a-mole with them to try and ensure compatibility. WebDAV is an old protocol, and support and implementation is far from universal across providers. It doesn't help that essentially all major vendors no longer support (or have never supported) this file syncing and sharing protocol.

In the upcoming update, M³ now stores media in a new way, in that no path needs to be configured by the user; everything "just works" out of the box. For this reason, those who would like to make use of a cloud sharing solution like Dropbox could use a workflow like either of the following ones.

### Workflow in current version of M³
- The _AV coordinator_ adds any needed additional media files to the cloud share from home using M³.
- After arriving at the KH, the _AV operator_ opens M³ on the media computer and waits for the sync to complete.

### New workflow 1
- The _AV coordinator_ uploads any needed additional media files to the cloud share from home.
- After arriving at the KH, the _AV operator_ opens the cloud share folder on the media computer.
- The AV operator downloads all files from the cloud folder.
- The AV operator drags and drops all downloaded files into the M³ window and reorders them as needed.

### New workflow 2
- The _AV coordinator_ creates a JWLPLAYLIST file with any needed additional media files, in the correct order.
- The AV coordinator uploads the JWLPLAYLIST file to the cloud share (ex.: Dropbox) from home.
- After arriving at the KH, the _AV operator_ opens the cloud share folder on the media computer.
- The AV operator downloads the JWLPLAYLIST file from the cloud folder
- The AV operator drags and drops the JWLPLAYLIST file into the M³ window. 

### New workflow 3
- After arriving at the KH, the _AV operator_ adds the song (for public talks) using the Import menu.
- The _AV operator_ drags and drops any additional pictures from a flash drive into the M³ window and reorders them as needed. 

In my testing, new workflow 2 is my favorite, as it is very quick to implement and does not take significantly more time than the workflow in the current version of M³. Realistically however, most congregations seem to already use new workflow 3 in practice, which (to be fair) works well too.

Hopefully this makes sense; I'm open to any feedback you might have.

**Reply by advenimus** [reply]
Dear @sircharlo ,

Thanks for getting back to us, we greatly appreciate your hard work on this. 

I believe this will be one of the most missed features in the new version, as the ability to remotely add media by our AV coordinator and enable auto sync when the AV operator for that meeting starts the program is incredibly simple. 

Is there no way to substitute use of webDAV with a different file share protocol such as FTP? There are tons of free FTP solutions out there, and all work the same way. This likely a more standardized than webDAV and will continue to be supported I'm sure for years to come.

I would love to help build something like this into M3 with you if needed 😄

**Reply by sircharlo** [reply]
The problem lies in the fact that before, M³ would download all meeting publications and media and copy relevant every single relevant media file into a dated folder for each meeting. The WebDAV sync would then simply add files to those dated media folders as needed. 

Now however, there will be no "dated media folders": all publication media is stored in a cached location, and the meeting view is essentially a list for every meeting date pointing to the various cache locations on disk. 

To add a syncing mechanism for custom media would mean that I'd need to build a system that:
1. Connects to a user-configurable (S)FTP server
2. Retrieves a list of media folders (somehow arranged by date?) and syncs those folders to disk somewhere (the cache folder?)
3. Injects whatever synced media was added since the last sync (new item: filename and checksum didn't exist before) into the relevant meetings
4. Loops over all injected media for every meeting and deletes references to any synced media that was deleted from the cloud (filename and checksum no longer exist in the cloud)
5. Deletes any synced media that is now orphaned (not referenced in media list) or outdated (for a past meeting) from the filesystem
6. Allow for custom sorting (somehow) 

These are not easy tasks. I'll accept any help that is offered though, if you feel like tackling this, even if only in part!

**Reply by Supamav** [reply]
Hello Brother Savignac,

Again, I can't tell you how much we appreciate your hard work, this program is incredibly versatile, and used heavily by our congregation and others in the area.

The ability to sync the files (whether WEBDAV or any type of syncing solution) in the background is a lifesaver when it comes to brothers who are not exactly 'Tech Savvy'.  Any of those workflows involving grabbing files from a file share, and importing them would present a significant hindrance to 90% of the brothers I have trained to use the program.  Most have difficulty remembering how to enter the password to get into the PC.  I suppose we could create a sort of template for them to use with either of the new workflows (1, 2).  Option 3 is not possible as we no longer can allow personal USB devices to connect to the KH PC.  Ask me how I know! :)

Syncing in the background (single pressing 'SYNC FOLDERS') is essential if for example I as the AV assistant coordinator can't be at the hall.

Can any of the following services be used to automatically sync?  Some allow for a 'mapped drive' and allow for access to folders, would it be possible to sync with what appears to be a local 'Folder'?  I don't mind myself, performing a few extra steps on the back-end to get the files in a position for the brothers to just sync at the hall:

Google Drive - A mapped drive can also be created here using Google Drive.
Dropbox
Onedrive
FileCloud (I host an instance at home) - In this use case a 'mapped drive' can be created to the instance of the Filecloud server for the KH PC.  Could we then have MMM sync with a folder on the 'mapped drive' or 'drive letter' in the same way we currently use WEBDAV shares?  See FileCloud Drive info here: https://www.filecloud.com/additional-downloads/

Thanks Again!
Jason

**Reply by sircharlo** [reply]
@Supamav @advenimus @e-Feitosa 
Since the vast majority of meeting media is already prepared and requires no intervention whatsoever, can I ask in what scenarios you are regularly needing to add/change meeting media?

Here are the scenarios I thought of: 

## Weekend Meeting
- #### Opening song
  In the new workflow, adding an opening song takes total of two clicks, typing in the song number, and then one final click. 
  **Time taken:** less than 10 seconds.
  **Frequency:** weekly
- #### Public talk media (from S-34mp or S-34 outline)
  In the new workflow, the S-34mp can be imported for easy use and re-use. 
  If the media comes from the S-34mp, adding it takes total of two clicks, finding the talk outline title or number, and then clicking on it. 
  If the media comes from an S-34 outline, it can be imported for easy use in one click by drag-and-drop, or by browsing in a total of two clicks, after which the user can browse to the JWPUB file. 
  **Time taken:** 10 to 30 seconds.
  **Frequency:** occasional
- #### Public talk media (from JWLPLAYLIST file)
  In the new workflow, a JWLPLAYLIST file can be imported for easy use in one click by drag-and-drop, or by browsing in a total of two clicks, after which the user can browse to the file.
  **Time taken:** less than 10 seconds.
  **Frequency:** occasional
- #### Public talk media (from other sources)
  In the new workflow, media files such as pictures can be imported for easy use in one click by drag-and-drop, or by browsing in a total of two clicks, after which the user can browse to the desired files.
  **Time taken:** 10 to 30 seconds.
  **Frequency:** occasional

## Midweek Meeting
- #### Additional media (from JWLPLAYLIST file)
  In the new workflow, a JWLPLAYLIST file can be imported for easy use in one click by drag-and-drop, or by browsing in a total of two clicks, after which the user can browse to the file.
  **Time taken:** less than 10 seconds.
  **Frequency:** rare
- #### Additional media (from other sources)
  In the new workflow, media files such as pictures can be imported for easy use in one click by drag-and-drop, or by browsing in a total of two clicks, after which the user can browse to the desired files.
  **Time taken:** 10 to 30 seconds.
  **Frequency:** rare

---

While I sincerely believe that adding the media (as in the scenarios I listed above) is part of the AV operator's responsibilities and is made very easy in the latest version of M³, I would of course like to hear your feedback. 
Thank you!

**Reply by e-Feitosa** [reply]
@sircharlo, Answering the question: "" The need is more on the weekend. Here in the region where I live, there are many small congregations and few brothers to take care of this designation. Many brothers have little help, and it makes it much easier to send the media before the weekend to get organized in the congregation and in case of travel, it helps a lot when the brother is absent. What the brother explained above, is sensational, but we have many audio operators who has a lot of difficulty in simpler things like downloading a file, dragging a file etc, may seem simple to most, but many have this difficulty mainly because they do not have a personal PC or Laptop and only have smartphone. One thing is certain, we will adapt, however, we will miss this feature. I use this feature of KOOFR, and has been virtually zero errors or problems in synchronization.

**Reply by sircharlo** [reply]
To make it even easier, I've added a "quick-add" button for WE meetings:

![image](https://github.com/user-attachments/assets/fdb32d8c-8b3e-4c51-8a6d-6721e9939902)

**Reply by advenimus** [reply]
Hello @sircharlo ,

The primary function that most AV coordinators (AVCs) utilize is the sync function, as it serves as the most convenient method for the brothers responsible for video to operate at the meeting. Few, if any, brothers actually know how to add the media, and this is going to require much further training, which will discourage some from the role, unfortunately. We will have to have designated brothers now to add the appropriate media as we do not have many "techy" brothers in our area.

Here is how many halls currently use M3. This also applies when midweek meetings require an adjusted program as per directions the BOE receives (ie. play the Governing Body Update in place of a local needs part).

## Current webDAV-based sync

- AVC gets an email containing info about the speakers’ talk, including song and images, days in advance. (more often than not, it's JPEG files and now .jwplaylist files)
- AVC then remotely adds this media to the webDAV server via M3 at home.
- When the Sunday meeting AV operator arrives, the program is opened, and a sync is forced right away.
- Media then automatically appears for the program that day, without intervention needed from the AV operator. 

This is, in my opinion, the most crucial function for M3, and it’s not just for us; many others rely on it as well. It would be very important to still have the ability to do something similar. 

If this is not being planned to be added to M3, then we will create some automations that will still just download media and add it to these cached folders you are speaking about. 

How can I go about getting a copy of the new version to test this with? Is it in the repo?

**Reply by advenimus** [reply]
Is it complicated to simply adapt the current webDAV implementation into the new version and state that is is an unsupported feature (ie. use at own risk, etc). I understand some tweaking would be necessary still as you've changed how folders are storing the media.

**Reply by e-Feitosa** [reply]
@advenimus,  
_Re: How can I go about getting a copy of the new version to test this with? Is it in the repo?_
https://github.com/sircharlo/mmm-refactor/releases/latest

**Reply by advenimus** [reply]
@e-Feitosa That's not the new version (25+)

**Reply by e-Feitosa** [reply]
@advenimus , This has not yet been published, I am also waiting.

**Reply by e-Feitosa** [reply]
@advenimus **v24.9.10**
https://github.com/sircharlo/mmm-refactor/releases/tag/v24.9.10

**Comment by advenimus** [comment]
Hi @sircharlo , can you please confirm the latest version without the major changes? We’ll probably need to use this moving forward. Thanks again for all your hard work!

**Reply by mtdvlpr** [reply]
Hi, `v24.4.x` will be the latest version before the big update. As long as the big update is not yet available, there might be small maintenance updates to this version. This release with its downloads will continue to be available for those who do not (yet) want to upgrade. You can disable auto updates in your settings to prevent accidentally updating.

After the official release of the big update, the old version won't receive any maintenance anymore.

**Comment by e-Feitosa** [comment]
So guys, any more ideas for a **WebDav** in the new version of M3?

**Reply by sircharlo** [reply]
If WebDAV is a dealbreaker, I would suggest staying at the current version and disabling auto-updates for now.
While I may explore this down the road, it's not high on my list of priorities at the moment.

**Reply by advenimus** [reply]
Thank you @sircharlo for your help. Will the banner stating final release be removed after new version is released? Or is that static forever for final version?

**Reply by advenimus** [reply]
Also, @sircharlo  can you provide some brief info on how your new version stores data on the local machine? (ie. what directory, how are specific meetings identified in the directory?) This may perhaps allow us to automate some synchronization via FTP to add the required media.

**Reply by e-Feitosa** [reply]
@advenimus , Did you get any alteranativa for the new version? My idea at the moment is to use Google Drive, but testing here, I have noticed that some will have quite a difficulty, especially when you send the file by mobile to a folder of Google Drive, it deletes the extension of the file. For example, when sending an image to a shared folder with the name image1.jpg, here it changes to the device name and removes the.JPG. Some have no difficulty in solving this, but others, simply can not put the file name and extension .JPG. The practicality of inserting media by WebDAV and download in KH is sensational. Any other practical suggestion you tested, share here with me doing a favor.

**Reply by advenimus** [reply]
@e-Feitosa 
Unfortunately I did not. However, I will be forking the old version final build and be stripping off some of the features such as Zoom integration and keep it to the basic OBS and WebDAV integration so that the most important features remain.

**Comment by e-Feitosa** [comment]
@sircharlo, **would it be possible to include an advanced and optional configuration in the M3 for it to detect media in a folder?** For example, when setting up **Dropbox/OneDrive/Google Drive** in KH and pointing this folder to M3, it would display a warning upon opening, indicating that there are additional media files, and ask if you want to import them. If it were possible to include this advanced option, it would solve the problem we will have with WebDAV.

**Reply by sircharlo** [reply]
Continuing this in #2767

**Reply by mtdvlpr** [reply]
I think you mean #2782

**Reply by sircharlo** [reply]
I'll mark this as awaiting release, since the folder watch feature is now implemented and will be released in the next point update!
